src/main/java/org/humancellatlas/ingest/analysis/web/AnalysisController.java

```diff
@@ -51,6 +53,25 @@
         return ResponseEntity.accepted().body(resource);
     }
 
+    @RequestMapping(path = "/submissionEnvelopes/{sub_id}/analyses/{id}", method = RequestMethod.PUT)
+    ResponseEntity<Resource<?>> linkAnalysisToEnvelope(@PathVariable("sub_id") SubmissionEnvelope submissionEnvelope,
+                                                      @PathVariable("id") Analysis analysis,
+                                                      final PersistentEntityResourceAssembler assembler) {
+        Analysis entity = getAnalysisService().addAnalysisToSubmissionEnvelope(submissionEnvelope, analysis);
+        PersistentEntityResource resource = assembler.toFullResource(entity);
+        return ResponseEntity.accepted().body(resource);
+    }
+
+    @RequestMapping(path = "/submissionEnvelopes/{sub_id}/analyses",
+            method = RequestMethod.PUT,
+            produces = MediaTypes.HAL_JSON_VALUE)
+    ResponseEntity<Resource<?>> linkAnalysesToEnvelope(@PathVariable("sub_id") SubmissionEnvelope submissionEnvelope,
+                                                     @RequestBody MetadataReference analysisReference,
+                                                     final PersistentEntityResourceAssembler assembler) {
+        SubmissionEnvelope entity = getAnalysisService().resolveAnalysisReferencesForSubmission(submissionEnvelope, analysisReference);
+        PersistentEntityResource resource = assembler.toFullResource(entity);
+        return ResponseEntity.accepted().body(resource);
+    }
```

Comment:
@aaclan-ebi is your archival code making lots of use of this style of endpoint? I think it's a little problematic, cos we're effectively making a single endpoint (`/submissionEnvelopes/{sub_id}/analyses`) have two different styles of behaviour depending on whether you send a `POST` or `PUT` request.

There's also a standard way to do this - see https://docs.spring.io/spring-data/rest/docs/current/reference/html/#repository-resources.association-resource.

If you really need this to make your code work, we can keep for now, but I much prefer the first style: `path = "/submissionEnvelopes/{sub_id}/analyses/{id}", method = RequestMethod.PUT`

Reply:
I see, okies, will change this. I also added a logic not to be able to link to a new submission env if it's still "open". Using the standard way may ignore that logic so I'll go with the first style.